src/manifest/mod.rs

```diff
@@ -343,3 +343,275 @@ pub mod provenance {
         const AUTHENTICATED: bool = false;
     }
 }
+
+wire_enum! {
+    /// A policy for responding to verification failure in a read-write region.
+    ///
+    /// Cerberus currently does not fully specify what these policies mean
+    /// precisely, nor what failure mode they should be enacted with respect
+    /// to.
+    pub enum RwFailurePolicy: u8 {
+        /// Do nothing.
+        DoNothing = 0b00,
+        /// Restore the region from write-only memory.
+        RestoreFromRo = 0b01,
+        /// Erase the region.
+        Erase = 0b10,
+    }
+}
+
+impl RwFailurePolicy {
+    /// Returns the `RwFailurePolicy` flag based on a `u8`
+    pub fn from_u8(n: u8) -> RwFailurePolicy {
```

Comment:
This is definitely interesting... at some point I may consider expanding the wire_enum! macro to simplify generation of bitfields like this one.

Reply:
yeah, we would need to allow that macro to take as input, the position of the flag in the enum.

So if there is another flag covering bits 5,6, the macro would be like:
```
// Yes the syntax is totally wrong, i'm not well read on macro syntax..
wire_bitflag! {
   position = 5..6,
   type = u8,
   enum Foo {
       bar = 0b00,
       ...
   }
}
```